cert/selfsigned.go

```diff
@@ -0,0 +1,172 @@
+package cert
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"math/big"
+	"net"
+	"os"
+	"time"
+)
+
+const (
+	// AutogenCertValidity is the default validity of a self-signed
+	// certificate. The value corresponds to 14 months
+	// (14 months * 30 days * 24 hours).
+	AutogenCertValidity = 14 * 30 * 24 * time.Hour
+)
+
+var (
+	// End of ASN.1 time.
+	endOfTime = time.Date(2049, 12, 31, 23, 59, 59, 0, time.UTC)
+
+	// Max serial number.
+	serialNumberLimit = new(big.Int).Lsh(big.NewInt(1), 128)
+)
+
+// GenCertPair generates a key/cert pair to the paths provided. The
+// auto-generated certificates should *not* be used in production for public
+// access as they're self-signed and don't necessarily contain all of the
+// desired hostnames for the service. For production/public use, consider a
+// real PKI.
+//
+// This function is adapted from https://github.com/btcsuite/btcd and
+// https://github.com/btcsuite/btcutil
+func GenCertPair(org, certFile, keyFile string, tlsExtraIPs,
+	tlsExtraDomains []string) error {
+
+	log.Infof("Generating TLS certificates...")
```

Comment:
Any reason we need these log messages to happen here? We could have them one layer above this (the caller of this function) and get rid of the `build` dependency on the module.

Reply:
I moved the log statements outside of the function and removed the logger in the `cert` package completely.
The only log statement that is now missing is the following, when the resolving of the local hostname fails:
```
		rpcsLog.Errorf("Failed getting hostname, falling back to "+
			"localhost: %v", err)
```
But I think we can live without that.